src/main/java/de/tum/cit/aet/artemis/iris/domain/settings/IrisLectureSettings.java

```diff
@@ -0,0 +1,89 @@
+package de.tum.cit.aet.artemis.iris.domain.settings;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.DiscriminatorValue;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToOne;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.cit.aet.artemis.core.domain.Course;
+
+/**
+ * An {@link IrisSettings} implementation for lecture specific settings.
+ * Lecture settings are used to override course settings and currently only allow setting the {@link IrisChatSubSettings}.
+ */
+@Entity
+@DiscriminatorValue("LECTURE")
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class IrisLectureSettings extends IrisSettings {
+
+    @OneToOne(optional = false)
+    @JoinColumn(name = "course_id", unique = true, nullable = false)
+    private Course course;
+
+    @OneToOne(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
+    @JoinColumn(name = "iris_chat_settings_id")
+    private IrisChatSubSettings irisChatSettings;
+
+    @OneToOne(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
+    @JoinColumn(name = "iris_lecture_chat_settings_id")
+    private IrisLectureChatSubSettings irisLectureChatSettings;
+
+    public Course getCourse() {
+        return course;
+    }
+
+    public void setCourse(Course course) {
+        this.course = course;
+    }
+
+    @Override
+    public IrisLectureIngestionSubSettings getIrisLectureIngestionSettings() {
+        return null;
+    }
+
+    @Override
+    public void setIrisLectureIngestionSettings(IrisLectureIngestionSubSettings irisLectureIngestionSettings) {
+    }
+
+    @Override
+    public IrisChatSubSettings getIrisChatSettings() {
+        return irisChatSettings;
+    }
+
+    @Override
+    public void setIrisChatSettings(IrisChatSubSettings irisChatSettings) {
+        this.irisChatSettings = irisChatSettings;
+    }
+
+    @Override
+    public IrisTextExerciseChatSubSettings getIrisTextExerciseChatSettings() {
+        return null;
+    }
+
+    @Override
+    public void setIrisTextExerciseChatSettings(IrisTextExerciseChatSubSettings irisTextExerciseChatSettings) {
+    }
+
+    @Override
+    public IrisLectureChatSubSettings getIrisLectureChatSettings() {
+        return irisLectureChatSettings;
+    }
+
+    @Override
+    public void setIrisLectureChatSettings(IrisLectureChatSubSettings irisLectureChatSettings) {
+        this.irisLectureChatSettings = irisLectureChatSettings;
+    }
+
+    @Override
+    public IrisCompetencyGenerationSubSettings getIrisCompetencyGenerationSettings() {
+        return null;
+    }
+
+    @Override
+    public void setIrisCompetencyGenerationSettings(IrisCompetencyGenerationSubSettings irisCompetencyGenerationSubSettings) {
+    }
```

Comment:
Same question here: is there a reason it sometimes returns null and the setter is doing nothing?